nbconvert/exporters/templateexporter.py

```diff
@@ -283,6 +284,8 @@ def _create_environment(self):
             extensions=JINJA_EXTENSIONS
             )
 
+        environment.globals['uuid4'] = uuid.uuid4
```

Comment:
It appears to work as is - I guess that Jinja does `str()` on objects, which gives you the same as `.hex`. This way feels neater, but I can also see the 'explicit is better than implicit' argument.